File: MSMA_Backend/src/main/java/com/spring/dto/response/ArtistPresentation.java
package com.spring.dto.response;

import com.spring.constants.UserType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ArtistPresentation {
    private Long id;
    private String avatar;
    private String firstName;
    private String lastName;
    private String artistName;
    private String description;
    private String image;
    private Long numberOfFollowers;
    private String email;
    private String gender;
    private String birthDay;
    private String phone;
    private Integer status;
    private Long createdBy;
    private Long lastModifiedBy;
    private String createdDate;
    private String lastModifiedDate;
    private UserType userType;
    private List<Long> artistSongIds;
    private List<String> artistSongNameList;
    private List<Long> artistPlaylistIds;
    private List<String> artistPlaylistNameList;
    private List<Long> artistAlbumIds;
    private List<String> artistAlbumNameList;

}

File: MSMA/app/src/main/java/vn/edu/usth/msma/data/dto/response/ArtistPresentation.kt
package vn.edu.usth.msma.data.dto.response

import vn.edu.usth.msma.utils.constants.UserType

data class ArtistPresentation(
    val id: Long,
    val avatar: String,
    val firstName: String,
    val lastName: String,
    val artistName: String,
    val description: String,
    val image: String,
    val backgroundImage: String,
    val countListen: Long,
    val numberOfFollowers: Long,
    val email: String,
    val gender: String,
    val birthDay: String,
    val phone: String,
    val status: Int,
    val createdBy: Long,
    val lastModifiedBy: Long,
    val createdDate: String,
    val lastModifiedDate: String,
    val userType: UserType
)
